src/app/rundown-view/services/tv2-piece-group.service.spec.ts

```diff
@@ -5,6 +5,80 @@ import { Tv2Piece } from '../../core/models/tv2-piece'
 import { TestEntityFactory } from '../../test/factories/test-entity.factory'
 
 describe(Tv2PieceGroupService.name, () => {
+  describe(Tv2PieceGroupService.prototype.mergePiecesByStartOffset.name, () => {
+    it('does not produce piece when no piece given', () => {
+      const pieces: Tv2Piece[] = []
+
+      const testee: Tv2PieceGroupService = createTestee()
+      const result: Tv2Piece[] = testee.mergePiecesByStartOffset(pieces)
+
+      const expected: Tv2Piece[] = []
+
+      expect(result).toEqual(expected)
+    })
+
+    it('does not alter piece given a single piece', () => {
+      const testEntityFactory: TestEntityFactory = new TestEntityFactory()
+
+      const pieceOne: Tv2Piece = testEntityFactory.createPiece({
+        name: 'pieceName',
+        start: 123,
+      }) as Tv2Piece
+      const pieces: Tv2Piece[] = [pieceOne]
+
+      const testee: Tv2PieceGroupService = createTestee()
+      const result: Tv2Piece[] = testee.mergePiecesByStartOffset(pieces)
+
+      const expected: Tv2Piece[] = [pieceOne]
+
+      expect(result).toEqual(expected)
```

Comment:
The test description and the implementation does not match. If `pieceOne` is modified within `mergePiecesByStartOffset`, then this test won't fail, because the expected output is based on the reference of the input. What you want to do is to create two instances of the pieces, which can be done by calling the `testEntityFactory.createPiece` twice or by spreading the instantiated piece.
The latter option does not take nested structures into account, so I would suggest using the test entity factory, e.g.:
```typescript
      const pieces: Tv2Piece[] = [
        testEntityFactory.createPiece({
          name: 'pieceName',
          start: 123,
        })
      ] as Tv2Piece[]
      const testee: Tv2PieceGroupService = createTestee()          

      const result: Tv2Piece[] = testee.mergePiecesByStartOffset(pieces)

      const expected: Tv2Piece[] = [
        testEntityFactory.createPiece({
          name: 'pieceName',
          start: 123,
        })
      ] as Tv2Piece[]
      expect(result).toEqual(expected)
```